Sources/CardPayments/CardClient.swift

```diff
@@ -7,27 +7,78 @@ import CorePayments
 public class CardClient: NSObject {
 
     public weak var delegate: CardDelegate?
-
+    public weak var vaultDelegate: CardVaultDelegate?
+    
     private let apiClient: APIClient
     private let config: CoreConfig
     private let webAuthenticationSession: WebAuthenticationSession
     private var analyticsService: AnalyticsService?
+    private var graphQLClient: GraphQLClient?
 
     /// Initialize a CardClient to process card payment
     /// - Parameter config: The CoreConfig object
     public init(config: CoreConfig) {
         self.config = config
         self.apiClient = APIClient(coreConfig: config)
         self.webAuthenticationSession = WebAuthenticationSession()
+        self.graphQLClient = GraphQLClient(environment: config.environment)
     }
 
     /// For internal use for testing/mocking purpose
-    init(config: CoreConfig, apiClient: APIClient, webAuthenticationSession: WebAuthenticationSession) {
+    init(
+        config: CoreConfig,
+        apiClient: APIClient,
+        webAuthenticationSession: WebAuthenticationSession,
+        graphQLClient: GraphQLClient? = nil
+    ) {
         self.config = config
         self.apiClient = apiClient
         self.webAuthenticationSession = webAuthenticationSession
+        self.graphQLClient = graphQLClient
     }
+    
+    public func vault(vaultRequest: CardVaultRequest) {
+        Task {
+            do {
+                let card = vaultRequest.card
+                let setupTokenID = vaultRequest.setupTokenID
+                let (updateResult) = try await updateSetupToken(vaultSetupTokenID: setupTokenID, card: card)
+                let result = updateResult
+                // TODO: handle 3DS contingency with helios link
+                if let link = result.links.first(where: { $0.rel == "approve" && $0.href.contains("helios") }) {
+                    let url = link.href
+                    print("3DS url \(url)")
+                } else {
+                    let vaultResult = CardVaultResult(setupTokenID: result.id, status: result.status)
+                    notifyVaultSuccess(for: vaultResult)
+                }
+            } catch let error as CoreSDKError {
+                notifyVaultFailure(with: error)
+            } catch {
+                notifyVaultFailure(with: CardClientError.vaultTokenError)
+            }
+        }
+    }
+    
+    func updateSetupToken(
+        vaultSetupTokenID: String,
+        card: Card
+    ) async throws -> TokenDetails {
```

Comment:
```suggestion
    func updateSetupToken(vaultSetupTokenID: String, card: Card) async throws -> TokenDetails {
```
I think this will be under our linter line length